src/wallet/wallet.cpp

```diff
@@ -137,6 +133,9 @@ void UnloadWallet(std::shared_ptr<CWallet>&& wallet)
     // Notify the unload intent so that all remaining shared pointers are
     // released.
     wallet->NotifyUnload();
+
+    wallet->m_chain_notifications_handler.reset();
```

Comment:
`BlockUntilSyncedToCurrentChain` is still needed, unless we can expect data loss

Reply:
re: https://github.com/bitcoin/bitcoin/pull/18338#discussion_r393264763

> `BlockUntilSyncedToCurrentChain` is still needed, unless we can expect data loss

BlockUntilSyncedToCurrentChain can't be called with locks held and it doesn't make sense here. It makes sense at beginning of RPC calls to ensure consistency between calls. It doesn't directly relate to writing or consolidating data.

Reply:
Where lock is held? If here it's, that's nasty bug.

Reply:
> Where lock is held? If here it's, that's nasty bug.

Right, I don't think there is a nasty bug now. I think it is good to avoid nasty bugs by not calling BlockUntilSyncedToCurrentChain when it is not necessary.